src/Engines/thinEngine.ts

```diff
@@ -5264,39 +5264,4 @@ export class ThinEngine {
         return document;
     }
 
-    /**
-     * Get Font size information
-     * @param font font name
-     * @return an object containing ascent, height and descent
-     */
-    public getFontOffset(font: string): { ascent: number, height: number, descent: number } {

```

Comment:
@CedricGuillemet had a reason to move it here, would be great to get his feedback on removing it.
(note - no back-compat issue as it was added after 5.0.0 work started)

Reply:
It's been added here for native compatibility with GUI: https://github.com/BabylonJS/Babylon.js/blob/057143356f8261a286b4af7a0a4844257373d27a/src/Engines/nativeEngine.ts#L2853
If GUI controls do not use this function any more, it's good to be removed from thinEngine and nativeEngine.

Reply:
@msDestiny14  - this implementation will fail on native, can you make sure that your function is also dealing with native context as well?

Reply:
Yup would be great if we can deal with it from GUI directly to not add extra code in Thin :-)

Reply:
I removed the function from the native side. The function inside of control.ts will now check if IsDocumentAvailable and will return 0's if not. (This was the native implementation)

@bghgary pinging you to keep you in the loop on the native side if you want to take a look.

Reply:
> I removed the function from the native side. The function inside of control.ts will now check if IsDocumentAvailable and will return 0's if not. (This was the native implementation)

I don't know if this will work. The native implementation is a placeholder. I won't guess that @CedricGuillemet intended on implementing it with some native call.

Reply:
In this case it might be moved on engine instead but not thinEngine ?

Reply:
I don't know this code that well, so let's get @CedricGuillemet's opinion first.

Reply:
I guess the main question is - does it need to be in the engine, or can it move back to the GUI and have a different implementation there? I doubt this is the only place where native and web differ, but I don't remember adding these kinds of functionalities to the engine.

Reply:
AFAIK, GUI implementation is agnostic between Native and web. Implementation differences only exist in Engine. It's possible to move this function in GUI and check if runtime is web or native. This means, a check and different code must exist there. It also means GUI knows about native implementation. Or only web version exists in GUI code and the function is overridden at runtime in native canvas. We can postpone this issue for now by moving the function in GUI, checking it's running on the web and return same object as that https://github.com/BabylonJS/Babylon.js/blob/ae488d1fc8d7599408a7e62ab877f19e32b8a0cb/src/Engines/nativeEngine.ts#L2855 otherwise. Canvas is not under active development for now so it won't be a problem for the weeks/months to come.

Reply:
We need to come up with a plan for platform abstraction. We have similar problems elsewhere and ideally we use the same pattern everywhere.

Reply:
I totally agree. Let's find time and discuss this.
Back to this PR - One of the two solutions will be enough:
1. @bghgary , @CedricGuillemet - is the current implementation enough for the current state in native? 
2. @msDestiny14 - is it possible to keep the current engine implementation and modify it there?

Reply:
Let's at least move it from ThinEngine to Engine :-)

Reply:
The check for document may not work in all cases since we sometimes fake a document to work around issues. Checking for `_native` should work though.

Reply:
So we are keeping it this way, but checking for _native instead of checking if document exists.